docs/design/proposals/logical-volume-manager-storage-api-v2.md

```diff
@@ -0,0 +1,591 @@
+            reasons:
+              - "/dev/sda1 was used as a boot partition and cannot be used"
+          - kname: "/dev/mapper/some-default-device"
+            reasons:
+              - "/dev/mapper/some-default-device is already being used by vg1 and cannot be used by a wildcard (Any) volumeGroup"
+          - kname: "/dev/mapper/some-default-device"
+            reasons:
+              - "/dev/mapper/some-default-device is already being used by vg1 and cannot be used by a wildcard (Any) volumeGroup"
+    - name: specific-serials
+      state: Healthy
+      size: "750Mi"
+      devices:
+        included:
+          - kname: "/dev/somedevwithserial"
+  deviceClasses:
+    - name: thinly-provisioned
+      volumeGroup: vg1
+      state: Healthy
+      storageClass:
+        name: "lvms-thinly-provisioned"
+        state: Exists
+      snapshotClass:
+        name: "lvms-thinly-provisioned"
+        state: Exists
+  conditions:
+    - lastTransitionTime: "2044-10-22T16:29:24Z"
+      status: "True"
+      type: VolumeGroupsHealthy
+    - lastTransitionTime: "2044-10-22T16:29:24Z"
+      status: "True"
+      type: DeviceClassesHealthy
+```
+
+Important to note is that any single deviceClass can only be mapped to one volumeGroup.
+Thus, all previous thin-pool configurations have the same limitation of being assignable to 1 volume group only.
+However, this should not be problematic as the `lvm2` system also strictly couples a thin-pool logical volume to a volume group, thus making this limit unimportant to the end user.
+
+### Implementation Details/Notes/Constraints
+
+#### When to deploy TopoLVM
+
+TopoLVM will be created once `LVMNodeSet` gets created based on the NodeSelector, just like `LVMCluster`.
+
+#### Constraints when creating LVMNode
+
+A validation webhook will prohibit creation of `LVMNode` outside of the scope of a `LVMNodeSet`.
+This is because within `LVMNodeSet`, the NodeSelector is used to create the DaemonSet for TopoLVM. If no NodeSet is present in the owner-references,
+the `LVMNode` will be rejected.
+
+Validation Pseudo-code:
+
+```
+nodeSetRef = first element in LVMNode.metadata.ownerReferences that contains LVMNodeSet and is controller-ref
+
+if nodeSetRef == nil
+    return error on create / update
+
+if object behind nodeSetRef.nodeSelector does not contain Node targeted by LVMNode
+    return error on create / update
+```
+
+#### Constraints when using multiple LVMNodeSets
+
+We suggest limiting LVMNodeSets to have singular ownership over any given node. That entails that one `LVMNode` is only able to be managed by one `LVMNodeSet`.
+Whenever an LVMNodeSet is created that would target a node that is already being covered by another `LVMNodeSet`, we suggest rejecting it through admission webhook.
+
+We also suggest binding the TopoLVM deployment / daemonset to a single `LVMNodeSet`, thus meaning that 2 `LVMNodeSet`'s would also deploy 2 instances of `TopoLVM` that should not be conflicting with each other.
```

Comment:
Do we know if it's possible to run two TopoLVM instances on a cluster? Could we maybe use `nodeSelector` on TopoLVM Daemonsets to run them only on the nodes specified by `LVMNodeSet`, so there will never be two instances running on the same node?